pkg/sql/catalog/systemschema/system.go

```diff
@@ -595,233 +653,180 @@ var (
 	// code assumes that it only has KV entries for column family 4, not the
 	// "sentinel" column family 0 which would be written by SQL.
-	NamespaceTable = makeTable(descpb.TableDescriptor{
-		Name:                    catconstants.NamespaceTableName,
-		ID:                      keys.NamespaceTableID,
-		ParentID:                keys.SystemDatabaseID,
-		UnexposedParentSchemaID: keys.PublicSchemaID,
-		Version:                 1,
-		Columns: []descpb.ColumnDescriptor{
-			{Name: "parentID", ID: 1, Type: types.Int},
-			{Name: "parentSchemaID", ID: 2, Type: types.Int},
-			{Name: "name", ID: 3, Type: types.String},
-			{Name: "id", ID: 4, Type: types.Int, Nullable: true},
-		},
-		NextColumnID: 5,
-		Families: []descpb.ColumnFamilyDescriptor{
-			{Name: "primary", ID: 0, ColumnNames: []string{"parentID", "parentSchemaID", "name"}, ColumnIDs: []descpb.ColumnID{1, 2, 3}},
-			{Name: "fam_4_id", ID: catconstants.NamespaceTableFamilyID, ColumnNames: []string{"id"}, ColumnIDs: []descpb.ColumnID{4}, DefaultColumnID: 4},
-		},
-		NextFamilyID: catconstants.NamespaceTableFamilyID + 1,
-		PrimaryIndex: descpb.IndexDescriptor{
-			Name:                "primary",
-			ID:                  catconstants.NamespaceTablePrimaryIndexID,
-			Unique:              true,
-			KeyColumnNames:      []string{"parentID", "parentSchemaID", "name"},
-			KeyColumnDirections: []descpb.IndexDescriptor_Direction{descpb.IndexDescriptor_ASC, descpb.IndexDescriptor_ASC, descpb.IndexDescriptor_ASC},
-			KeyColumnIDs:        []descpb.ColumnID{1, 2, 3},
-		},
-		NextIndexID: 2,
-		Privileges: descpb.NewCustomSuperuserPrivilegeDescriptor(
-			descpb.SystemAllowedPrivileges[keys.NamespaceTableID], security.NodeUserName(),
-		),
-		FormatVersion:  descpb.InterleavedFormatVersion,
-		NextMutationID: 1,
-	})
+	NamespaceTable = toCatalogDescriptor(
+		systemTable(
+			catconstants.NamespaceTableName,
+			keys.NamespaceTableID,
+			[]descpb.ColumnDescriptor{
+				{Name: "parentID", ID: 1, Type: types.Int},
+				{Name: "parentSchemaID", ID: 2, Type: types.Int},
+				{Name: "name", ID: 3, Type: types.String},
+				{Name: "id", ID: 4, Type: types.Int, Nullable: true},
+			},
+			[]descpb.ColumnFamilyDescriptor{
+				{Name: "primary", ID: 0, ColumnNames: []string{"parentID", "parentSchemaID", "name"}, ColumnIDs: []descpb.ColumnID{1, 2, 3}},
+				{Name: "fam_4_id", ID: catconstants.NamespaceTableFamilyID, ColumnNames: []string{"id"}, ColumnIDs: []descpb.ColumnID{4}, DefaultColumnID: 4},
+			},
+			descpb.IndexDescriptor{
+				Name:                "primary",
+				ID:                  catconstants.NamespaceTablePrimaryIndexID,
+				Unique:              true,
+				KeyColumnNames:      []string{"parentID", "parentSchemaID", "name"},
+				KeyColumnDirections: []descpb.IndexDescriptor_Direction{descpb.IndexDescriptor_ASC, descpb.IndexDescriptor_ASC, descpb.IndexDescriptor_ASC},
+				KeyColumnIDs:        []descpb.ColumnID{1, 2, 3},
+			},
+		))
```

Comment:
This should yield the same descriptor, but more concisely and with less potential for bugs. Recall that these descriptors are all tested in `TestSystemTableLiterals`.